### PR TITLE
Add support for `pomFile` parameter

### PR DESCRIPTION
**Pull request by smaarn**
### Issue

The value of `mavenProject.getFile()` may be referencing a file which has been altered (for example when using the flatten maven plugin) depending on when it's invoked.

The idea of this MR is to allow the explicit specification of the POM file to format (or check) by making it possible to pass a custom value to `pomFile`.

Would allow to fix https://github.com/mojohaus/tidy-maven-plugin/issues/72 (by making it explicit about the pom and not the "potentially altered file")

### CODE REVIEW

**Comment by smaarn** [comment]
It should be ok to review now.

The only question I had would be whether or not I should update the README accordingly also. 🤷

**Comment by slawekjaranowski** [comment]
> It should be ok to review now.
> 
> The only question I had would be whether or not I should update the README accordingly also. 🤷

Documentation on new parameter is enough, we not need update readme.

File: src/main/java/org/codehaus/mojo/tidy/TidyMojo.java [hunk 1]
```diff
@@ -46,6 +46,14 @@ public abstract class TidyMojo extends AbstractMojo {
     @Parameter(defaultValue = "${project}", required = true, readonly = true)
     protected MavenProject project;
 
+    /**
+     * The path of the pom file to process.
+     *
+     * @since 1.3.0
```
Comment (slawekjaranowski) [inline src/main/java/org/codehaus/mojo/tidy/TidyMojo.java:52]: Please also update project to version 1.3.0-SNAPSHOT ... we can forget about it during release

Comment (smaarn) [reply src/main/java/org/codehaus/mojo/tidy/TidyMojo.java:52]: TBH I didn't know what was the recommended procedure. Done.

**Review by slawekjaranowski — APPROVED** [review]

**Pull request merged**